ui/src/pages/domain/[domain]/service/[service]/tags.js

```diff
@@ -0,0 +1,222 @@
+import { ReduxPageLoader } from '../../../../../components/denali/ReduxPageLoader';
+import {
+    selectService,
+    selectServiceTags,
+} from '../../../../../redux/selectors/services';
+import ServiceTabs from '../../../../../components/header/ServiceTabs';
+import { getServices } from '../../../../../redux/thunks/services';
+
+const AppContainerDiv = styled.div`
+    align-items: stretch;
+    flex-flow: row nowrap;
+    height: 100%;
+    display: flex;
+    justify-content: flex-start;
+`;
+
+const MainContentDiv = styled.div`
+    flex: 1 1 calc(100vh - 60px);
+    overflow: hidden;
+    font: 300 14px HelveticaNeue-Reg, Helvetica, Arial, sans-serif;
+`;
+
+const TagsContainerDiv = styled.div`
+    align-items: stretch;
+    flex: 1 1;
+    height: calc(100vh - 60px);
+    overflow: auto;
+    display: flex;
+    flex-direction: column;
+`;
+
+const TagsContentDiv = styled.div``;
+
+const PageHeaderDiv = styled.div`
+    background: linear-gradient(to top, #f2f2f2, #fff);
+    padding: 20px 30px 0;
+`;
+
+export async function getServerSideProps(context) {
+    let api = API(context.req);
+    let reload = false;
+    let notFound = false;
+    let error = null;
+    const tagsData = await Promise.all([api.getForm()]).catch((err) => {
+        let response = RequestUtils.errorCheckHelper(err);
+        reload = response.reload;
+        error = response.error;
+        return [{}];
+    });
+    return {
+        props: {
+            reload,
+            notFound,
+            error,
+            serviceName: context.query.service,
+            userName: context.req.session.shortId,
+            _csrf: tagsData[0],
+            domainName: context.query.domain,
+            nonce: context.req.headers.rid,
+        },
+    };
+}
+
+class ServiceTagsPage extends React.Component {
+    constructor(props) {
+        super(props);
+        this.api = API();
+        this.cache = createCache({
+            key: 'athenz',
+            nonce: this.props.nonce,
+        });
+        this.state = {
+            error: null,
+            reload: false,
+        };
+    }
+
+    componentDidMount() {
+        const { domainName, userName, getDomainData, getServices } = this.props;
+        Promise.all([
+            getDomainData(domainName, userName),
+            getServices(domainName),
+        ]).catch((err) => {
+            let response = RequestUtils.errorCheckHelper(err);
+            this.setState({
+                error: response.error,
+                reload: response.reload,
+            });
+        });
+    }
+
+    render() {
+        const {
+            domainName,
+            reload,
+            serviceName,
+            serviceDetails,
+            serviceTags,
+            _csrf,
+            isLoading,
+        } = this.props;
+        if (reload || this.state.reload) {
+            window.location.reload();
+            return <div />;
+        }
+        const err = this.props.error || this.state.error;
```

Comment:
Minor: you can do this here for cleaner code
```
serviceDetails = serviceDetails || {};
```
then replace lines 161, 168,  181... with just
```
collectionDetails={serviceDetails}
```

Reply:
serviceDetails is a const because it does not suppose to be changed on the page, but I took your approach and change the lines to serviceDetails || {} which is more elegant.